chapter/src/main/java/com/nk/DetailsRepository.java

```diff
@@ -75,14 +95,219 @@ public  boolean login (UserDetails user) {
 
             System.out.println(rs);
             if(rs.next()){
-            flag= true;
+                cookieid = -1;
+                rs = st.executeQuery(sql1);
+                if(rs.next())
+                {cookieid= rs.getInt("userid");}
             }
 
         } catch (SQLException e) {
             // TODO Auto-generated catch block
             e.printStackTrace();
         }
 
-        return flag ;
+        return cookieid ;
+    }
+
+
+    public  void addUser ( String email,int adminId) {
+
+        String sql = " Insert into invites  values(?,?)";
+        try {
+            if( mailService.sendMail(email)){
```

Comment:
separate the functionalities. Don't mix email sending in a class where you do DB ops.